src/main/java/org/aludratest/service/gui/web/WebGUIInteraction.java

```diff
@@ -101,10 +103,22 @@ String clickForDownload(@ElementType String elementType, @ElementName String ele
      * The framework may or may not be supported by the web GUI implementation. If it is not supported, an AutomationException is
      * thrown. <br>
      * If the maximum waiting time is reached without the Ajax operation being finished, a PerformanceException is thrown.
-     * 
+     *
      * @param frameworkName Name of the JavaScript framework to check, e.g. "jquery". Implementations should convert this
      *            parameter to lowercase before checking it, so case does not matter for the caller.
      * @param maxWaitTime Maximum time, in milliseconds, to wait for the AJAX operation to be finished. */
     void waitForAjaxOperationEnd(@TechnicalArgument String frameworkName, @TechnicalArgument int maxWaitTime);
 
+    /** Adds or creates a cookie.
+     * @param name cookie name
+     * @param value cookie value
+     * @param domain string
+     * @param path string
+     * @param expiry date */
+    void addCookie(@TechnicalLocator String name, @TechnicalArgument String value, @TechnicalArgument String domain,
+            @TechnicalArgument String path, @TechnicalArgument Date expiry);
```

Comment:
I'd appreciate an int parameter here for the number of seconds the cookie shall be valid (instead of a Date), which can then internally be calculated to a Date (in the implementation in aludratest-selenium). We try to keep only very simple parameter types here.